### PR TITLE
feat(subtasks): add the Subtasks service and repin to bc3 5daa0911d3

### PR DESCRIPTION
**Pull request by zachasme**
bc3 #12659 documented subtasks as a first-class resource with flat canonical routes, shaped after comments: a to-do or card advertises `subtasks_count`, `subtasks_completed_count` and `subtasks_url`, and the index paginates. This models it as a `Subtasks` service in all seven SDKs — list, get, create, update, complete, uncomplete, reposition, delete — reusing the `CardStep` structure, because the wire shape did not change: `type` stays `"Kanban::Step"`, only the routes and the parent's accounting are new. `Todo`, `Card` and the generic `Recording` carry the three accounting members. `CardSteps` stays as it was; bc3 keeps the card-scoped `/steps` spellings served indefinitely as aliases of the same records.

Absorbing documented routes means moving the provenance pin, so this advances it from `c680233ba0e` to `5daa0911d33` and triages the range in `spec/api-gaps/README.md`. Client users had already been absorbed ahead of the repin. The three other documented additions — the unscoped recording show (#10158), backlinks (#13121) and bulk enrollment (#9962) — are registered as `addressed-in-bc3-pr` briefs for follow-up absorption rather than modelled here, with matching dispositions in the route allowlist, and the client-route waivers the repin makes redundant are deleted.

Two corrections rode along and are the reason for the `breaking` label. bc3's `card_table_steps.md` called the reposition `position` "Zero indexed" until #12659 fixed it to 1-based; the server always counted from 1, so `RepositionCardStep`'s member doc and Go's `CardStepsService.Reposition` lower bound now say so (MIGRATING has the note). And `PUT /my/unreads` refuses batches over 500 readables with a 422, so `MarkAsRead` declares `ValidationError` and a `@length(max: 500)`.

### CODE REVIEW

**Comment by zachasme** [comment]
Verification notes, for the reviewer rather than the release notes.

`make -k check-targets` ran locally with every gate green except two the machine cannot run: `swift-check` (the Swift package uses CoreFoundation and builds only on macOS; the generator itself ran in a `swift:6.1` container and its output is what is committed, so `swift-check-drift` in CI is the real verification) and `rs-deny` (no `cargo-deny` installed). Kotlin built and tested through Gradle, the conformance runners replayed 215 cases, TypeScript 1750 tests, Ruby 1537 runs, Go, Python and Rust suites all pass, and route parity reports 270 SDK routes against 398 bc3 routes at the new pin.

Design notes. The shape is deliberately still `CardStep`: renaming it `Subtask` across seven SDKs is a breaking rename with no wire change behind it, so it is a separate decision. The Go wrapper's `Update` hand-marshals its body the way `CardStepsService.Update` does, because `"due_on": ""` is the only spelling of a clear the generated `*types.Date` cannot express (SPEC §18 rule 1). `DefaultSubtaskLimit` is 100 to match the 100 a parent embeds under `steps`, so a default listing never returns less than the parent already showed.

The range triage counted eleven non-merge commits touching `doc/api` or `app/views/api`; the table grew by exactly the sixteen routes the entries above account for. Circle creation (#9949) gained API views but no documentation, so it is recorded in the triage and not registered — a documented contract is the trigger.

**Comment by zachasme** [comment]
Two red checks are inherited from `main`, not this branch: `cargo deny (Rust SDK)` and the `Rust stable (1.98.1)` job both stop at the cargo-deny stage on RUSTSEC-2026-0285 (a rustls TLS 1.3 advisory in a transitive dependency). `main` at f2f900f6 fails the same two, and this branch touches no Rust dependencies; the Rust fmt, lint, tests and docs stages pass before deny runs. A dependency bump clears both, and belongs in its own PR.

**Comment by zachasme** [comment]
Codex adversarial review (read-only, against the local bc3 checkout at 5daa0911d3) and what came of it, folded into the one commit:

Fixed. Go's `Reposition` in both `SubtasksService` and `CardStepsService` cast `int` to `int32` after checking only the lower bound, so `2147483648` would have wrapped negative and landed the item at the top instead of the bottom. Both now refuse anything above `math.MaxInt32`, with a test. The `account-people-enrollment` brief's suggested error list omitted the documented `507 Insufficient Storage` for the account user limit, which bc3 raises from both the preflight and the enrollment path; the brief now names it, modelled like `StorageLimitError`.

Also from CONTRIBUTING, not Codex: the commit and title now follow Conventional Commits, and the root, TypeScript, Ruby and Go READMEs gained the `Subtasks` row their service tables were missing.

Noted, not changed. Rust cannot express bc3's due-date clear (`null` or `""`) on `UpdateSubtask` because the generated `Option<Date>` omits on `None` — the same standing limitation `UpdateCardStep` and every generated date field carry, and the reason the merge-safe composites exist; a Rust composite for subtasks would be a separate PR. Go's `UpdateSubtaskRequest.Title` treats `""` as "leave unchanged", which mirrors `UpdateStepRequest` and is documented on the field; a pointer would let a caller send an empty title on purpose, but I kept the two request types symmetrical.

Not done: CONTRIBUTING step 5 asks for conformance fixtures for new operations. Neither `Boosts` nor `CardSteps` has one today, and a new fixture means dispatch arms in all seven runners, including Swift, which this machine cannot compile. Happy to add one if you want it in this PR.

File: go/pkg/basecamp/subtasks.go [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+	ctx = s.client.parent.hooks.OnOperationStart(ctx, op)
+	defer func() { s.client.parent.hooks.OnOperationEnd(ctx, op, err, time.Since(start)) }()
+
+	if position < 1 {
+		err = ErrUsage("position must be at least 1")
+		return err
+	}
+
+	body := generated.RepositionSubtaskJSONRequestBody{
+		Position: int32(position), // #nosec G115 -- position is validated and bounded by API
```
[the remark this answers is not in the document]
Comment (zachasme) [reply go/pkg/basecamp/subtasks.go]: 🤖 Already handled before this review posted: subtasks.go now checks `position > math.MaxInt32` before the int32 cast, matching the todolist wrapper. The cast site carries the nosec note. 45319b11 also rewrites the range test so MaxInt32+1 is built at runtime and compiles on 32-bit.

File: python/tests/services/test_subtasks_service.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        "assignees": [],
+        "completion_url": f"{BASE}/subtasks/{subtask_id}/completion.json",
+    }
+
+
+def _subtasks():
+    return Client(access_token="test-token").for_account("12345").subtasks
+
+
+class TestList:
```
[the remark this answers is not in the document]
Comment (zachasme) [reply python/tests/services/test_subtasks_service.py:48]: 🤖 Added in 45319b11: update 422 → ValidationError, uncomplete 404 → NotFoundError, reposition 422 → ValidationError.

File: ruby/test/basecamp/services/subtasks_service_test.rb [hunk 3]
```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+# Tests for the SubtasksService (generated from OpenAPI spec)
+#
+# A subtask is a CardStep on the wire — `type` stays "Kanban::Step" — reached
+# through the canonical flat /subtasks routes bc3 documents (bc3#12659).
+
+require "test_helper"
+
+class SubtasksServiceTest < Minitest::Test
```
[the remark this answers is not in the document]
Comment (zachasme) [reply ruby/test/basecamp/services/subtasks_service_test.rb:10]: 🤖 Added in 45319b11: get 404, update 422, complete 404, uncomplete 404, reposition 422, one assertion each.

File: typescript/tests/services/subtasks.test.ts [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+  completed: false,
+  due_on: null,
+  parent: { id: 200, title: "Shot list", type: "Todo", url: `${BASE_URL}/buckets/1/todos/200.json`, app_url: "https://3.basecamp.com/12345/buckets/1/todos/200" },
+  bucket: { id: 1, name: "The Leto Laptop", type: "Project" },
+  creator: { id: 100, name: "Matt Donahue" },
+  assignees: [],
+  completion_url: `${BASE_URL}/subtasks/${id}/completion.json`,
+});
+
+describe("SubtasksService", () => {
```
[the remark this answers is not in the document]
Comment (zachasme) [reply typescript/tests/services/subtasks.test.ts:37]: 🤖 Added in 45319b11: update 422, complete 404, uncomplete 404, reposition 422.

**Review by robzolkos — CHANGES_REQUESTED** [review]
🤖 _This review was drafted with AI assistance._

Hey Zach, this is a really solid piece of work. You modelled all eight operations across seven SDKs, absorbed the repin and triaged its range, and handled the Copilot feedback cleanly. The gap brief write-up and the verification notes made this much easier to review. Thank you.

I checked the routes, status codes and parameters against bc3's `subtasks.md` and the controller at `5daa0911d33`, and they all line up. A few small things before I approve:

**1. A fixture for the new `Recording` fields**

`subtasks_count`, `subtasks_completed_count` and `subtasks_url` are new on the generic `Recording`, but neither of the manifest's `Recording` examples includes them. One is a Message, the other a Todolist. We hit the same thing with `bubble_up_url`, and the `recording-list-todolist` entry is the pattern that fixed it. Could you add a to-do- or card-shaped element that carries all three keys and register it in `spec/fixtures/manifest.yaml`? Then `check-fixture-coverage` will guard those fields like the rest.

**2. `due_on` clear wording on `UpdateSubtask`**

The doc comment says to send `"due_on": ""` to clear the date. That works: `SubtasksController` passes it to `assign_attributes` on a `date` column, which casts `""` to nil. But bc3 documents `"due_on": null` for this endpoint, and that's also what its test covers ("API: update with an explicit null due date clears it"). Since this comment ends up in every SDK's docs, could it lead with `null` and mention `""` as also accepted? Something like:

> Clearing a value takes an explicit send — `"due_on": null` clears the due date (an empty string is also accepted), `"assignee_ids": []` removes every assignee.

**3. The `breaking` label**

I think the label is fair, but the description makes it sound bigger than it is. The `MarkAsRead` change adds `@length(max: 500)` and a `ValidationError`, but no generator enforces the length on the client. So callers get the same 422 from the server they always would have, and that part only changes documentation. The one real behaviour change is Go's `CardStepsService.Reposition` rejecting `0`. Could you tweak the description so someone skimming the release notes knows that's the only thing to check?

Two other things came up during review that aren't from this PR, so I've filed them separately instead of adding to your list:
- `CardStep` doesn't model the `completion` object bc3 sends for completed steps and subtasks. Its `completed_at` and `completer` fields are never populated.
- The Rust generator can't send an explicit `null` on optional request fields, so Rust users can't clear a due date on any update. `UpdateSubtask` just inherits that.

Neither needs to block this.

Thanks again, this is close. Happy to re-review as soon as those land. 🙏

**Comment by zachasme** [comment]
Thanks for reviewing, @robzolkos 🙏 I believe your 3 points are addressed in https://github.com/basecamp/basecamp-sdk/pull/883/changes/a3310d5a8c727f1c640289042154c60ad92aa7d6 (and the updated PR body).
